### PR TITLE
Issue #1575: Make disabled filter formats enableable.

### LINKED ISSUE #1575: [UX] Text formats: provide a way to re-enable disabled formats via the admin UI
A 'Disabled' text format is different than every other 'Disabled' thing in the admin Ui. A text format, once disabled, can not be enabled again. Everything else can. This can be a pain-point for admins who think they know what to expect.

Possible solutions that have been considered:

1) Allow admins to re-enable disabled text formats, making the operation reversible.
2) Change the name from 'Disabled' to something more like 'Hidden'
3) Allow admins to completely 'Delete' a text format (possibly dangerous!)

The final approach taken in this issue was to keep disabled formats in the text formats list in `/admin/config/content/formats`, denote them as disabled, and allow for them to be re-enabled (via a newly-introduced "Enable" operation). See https://github.com/backdrop/backdrop-issues/issues/1575#issuecomment-527534721 for a screenshot of this implementation.

This fixes the actual UX issue here, which is that previously we were telling people that we are going to disable the text format, but what we were actually doing instead was to completely remove it from the UI, leaving the user wondering how to recover from this situation. Advanced users knew how to revert this, by manually editing the config files; the new implementation makes it so that even novice users can accomplish the same thing via the admin UI.

The actual "Remove" functionality, and whether we should implement that as complete removal of the text format config files, or simply hide them from the admin UI and keep the config files in place, was left as a follow-up task in #4013.

---

## Original Issue

Most site admins think that something "disabled" can be "enabled" again.
Ok, there is a confirm screen, but not everybody read them, especially if they are written in a foreign language (and drupal-7.41.##.po does not include text of `admin/config/content/formats/{format}/disable`)...

----

Advocate:  @klonos

----

Weekly update?

----

~~PR https://github.com/backdrop/backdrop/pull/2821~~ by @BWPanda
~~PR https://github.com/backdrop/backdrop/pull/2826~~ by @klonos (based on @BWPanda's original PR): 
~~PR https://github.com/backdrop/backdrop/pull/2845~~ by @quicksketch 
~~PR https://github.com/backdrop/backdrop/pull/2846 by @jenlampton~~
~~PR  https://github.com/backdrop/backdrop/pull/3024~~ by @klonos
PR https://github.com/backdrop/backdrop/pull/3071 by @docwilmot

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/1575
Replaces https://github.com/backdrop/backdrop/pull/2826

### CODE REVIEW

File: core/modules/filter/filter.admin.inc [hunk 1]
```diff
@@ -14,7 +14,7 @@
  */
 function filter_admin_overview($form) {
   // Overview of all formats.
-  $formats = filter_formats();
+  $formats = filter_formats(NULL, TRUE, TRUE);
```
Comment (quicksketch) [inline core/modules/filter/filter.admin.inc]: The last `TRUE` can be removed.
```suggestion
  $formats = filter_formats(NULL, TRUE);
```

Comment (jenlampton) [reply core/modules/filter/filter.admin.inc]: yeah, this was the one that came with the patch I made from your PR. Sorry about that, caught it and still forgot to delete...  killed now.

File: core/modules/filter/filter.admin.inc [hunk 2]
```diff
@@ -14,10 +14,15 @@
...
 function filter_admin_overview($form) {
   // Overview of all formats.
-  $formats = filter_formats();
+  $formats = filter_formats(NULL, TRUE);
   $fallback_format = filter_fallback_format();
   $editors = filter_get_editors();
 
+  $form['help'] = array(
+    '#type' => 'help',
+    '#markup' => t('The order of the text formats below determines the order in which they will appear in the <em>Editor</em> selector when creating content (depending on permissions for the text formats.'),
```
Comment (klonos) [inline core/modules/filter/filter.admin.inc]:
```suggestion
    '#markup' => t('The order of the text formats below determines the order in which they will appear in the <em>Editor</em> selector when creating content (depending on permissions for the text formats).'),
```

**Comment by jenlampton** [comment]
closing for new sandbox.

**Comment by jenlampton** [comment]
new sandbox please.

**Comment by jenlampton** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/3071

**Comment by jenlampton** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/3071

**Pull request closed without merging**